tests/Alcaeus/MongoDbAdapter/Mongo/MongoClientTest.php

```diff
@@ -9,6 +9,24 @@
  */
 class MongoClientTest extends TestCase
 {
+    /**
+     * @dataProvider provideConnectionUri
+     */
+    public function testConnectionUri($uri, $expected)
+    {
+        $this->assertSame($expected, (string) (new \MongoClient($uri, ['connect' => false])));
+    }
+
+    public function provideConnectionUri()
+    {
+        yield ['default', sprintf('mongodb://%s:%d', \MongoClient::DEFAULT_HOST, \MongoClient::DEFAULT_PORT)];
+        yield ['localhost', 'mongodb://localhost'];
+        yield ['mongodb://localhost', 'mongodb://localhost'];
+        if (version_compare(phpversion('mongodb'), '1.4.0', '>=')) {
+            yield ['mongodb+srv://localhost', 'mongodb+srv://localhost'];
```

Comment:
libmongoc requires that there by only a single hostname and that the hostname string contain at least two dots (https://github.com/mongodb/mongo-c-driver/blob/30085af2124de68fd5ad714309f3ea2688489dee/src/mongoc/mongoc-uri.c#L430). I'm not sure if requiring two dots is correct, as the [specification](https://github.com/mongodb/specifications/blob/master/source/initial-dns-seedlist-discovery/initial-dns-seedlist-discovery.rst#specification) only suggests a hostname and domain suffix. I'll follow up if this turns out to be a libmongoc bug, but let's use `foo.example.com` for now.

Reply:
Ah, this actually isn't a bug. I misread the spec. It requires `{hostname}.{domainname}`, where a domain name includes its own dot before the TLD suffix.

There's also a test for this in [not-enough-parts.json](https://github.com/mongodb/specifications/blob/master/source/initial-dns-seedlist-discovery/tests/not-enough-parts.json).

Reply:
Thanks a lot for finding this!